[기송은] Week5

q45402sk opened the pull request:
## 요구사항

- [x] [기본]이메일 input에서 focus out 할 때, 값이 없을 경우 input에 빨강색 테두리와 아래에 “이메일을 입력해주세요.” 빨강색 에러 메세지가 보이나요?
- [x] [기본]이메일 input에서 focus out 할 때, 이메일 형식에 맞지 않는 값이 있는 경우 input에 빨강색 테두리와 아래에 “올바른 이메일 주소가 아닙니다.” 빨강색 에러 메세지가 보이나요?
- [x] [기본]이메일 input에서 focus out 일 때, input 값이 test@codeit.com 일 경우 input에 빨강색 테두리와 아래에 “이미 사용 중인 이메일입니다.” 빨강색 에러 메세지가 보이나요?
- [x] [기본]비밀번호 input에서 focus out 할 때, 값이 8자 미만으로 있거나 문자열만 있거나 숫자만 있는 경우, input에 빨강색 테두리와 아래에 “비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.” 빨강색 에러 메세지가 보이나요?
- [x] [기본]비밀번호 input과 비밀번호 확인 input의 값이 다른 경우, 비밀번호 확인 input에 빨강색 테두리와 아래에 “비밀번호가 일치하지 않아요.” 빨강색 에러 메세지가 보이나요?
- [x] [기본]회원가입을 실행할 경우, 문제가 있는 경우 문제가 있는 input에 빨강색 테두리와 에러 메세지가 보이나요?
- [x] [기본]회원가입 버튼 클릭 또는 Enter키 입력으로 회원가입 되나요?
- [x] [기본]이메일, 비밀번호, 비밀번호 확인 input에 에러 관련 디자인을 Components 영역의 에러 케이스로 적용했나요?
- [x] [심화]눈 모양 아이콘 클릭시 비밀번호의 문자열이 보이기도 하고, 가려지기도 하나요?
- [x] [심화]비밀번호의 문자열이 가려질 때는 눈 모양 아이콘에는 사선이 그어져있고, 비밀번호의 문자열이 보일 때는 사선이 없는 눈 모양 아이콘이 보이나요?
- [x] [심화]로그인, 회원가입 페이지에 공통적으로 사용하는 로직이 있다면, 반복하지 않고 공통된 로직을 모듈로 분리해 사용했나요?

## 주요 변경사항

- 회원가입 페이지 에러 메세지와 빨간색 테두리 추가
- 로그인, 회원가입 페이지의 공통된 로직을 모듈로 분리해 사용


## 멘토에게

-
-
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

q45402sk committed:
폴더 추가

q45402sk committed:
폴더 내 파일 추가

q45402sk committed:
폴더 내 파일 추가1

q45402sk committed:
classname 수정

q45402sk committed:
모바일 화면에서 이미지가 보이지 않는 현상 수정

q45402sk committed:
Merge branch 'part1-기송은' of https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission into part1-기송은

q45402sk committed:
Merge branch 'part1-기송은' of https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission into part1-기송은

q45402sk committed:
fix: Enter키를 눌렀을 때 로그인이 안되는 현상 수정

q45402sk committed:
"refactor: 4주차 코드리뷰 반영"

-signin.html: error message component 하나로 줄이기

q45402sk committed:
feat: 회원가입 페이지 업데이트

5주차 위클리 미션
-회원가입: 에러메세지 업데이트
-유효한 회원가입 시도의 경우 /folder로 이동

q45402sk committed:
feature: 에러메세지일 때 빨간색 테두리 추가

1005hoon commented on signin/common.js:
```diff
@@ -0,0 +1,54 @@
+    eyeoffImg.setAttribute("src", "../public/images/signin-image/eye-off.png");
+    showPasswordButton[i].prepend(eyeoffImg);
+  }
+}
```
요것두 `togglePasswordVisibility` 라는 함수 내에서 `showPassword` `hidePassword` 함수를 호출하는 식으로 분리해보면 어떨까요?

1005hoon commented on signin/index.html:
```diff
@@ -39,12 +32,7 @@
-            <!-- 비밀번호를 확인해 주세요. -->
-            <p class="check-pw"></p>
-          </div>
+          <p class="pw-error-message"></p>
```
에러메시지들이 지녀야 할 공통 UI 스타일이 있을텐데요
이런 스타일들을 `error-message` 라는 클래스로 두고, 각 에러메시지에 대한 선택을 `id=pw-error` 또는 `id=email-error` 로 구분해 활용할 수 있도록? 해보면 어떨까요

1005hoon commented on signin/index1.js:
```diff
@@ -16,32 +22,22 @@ loginButton.addEventListener("click", function (event) {
+    event.preventDefault();
+    handleLogin(event);
+  }
+});
```
form 내부에 type=submit 버튼이 있으면 폼 내부에서 엔터를 쳤을 때 `onSubmit` 액션이 트리거될텐데요, 
이 때 로그인 로직이 실행되도록 해보면 어떨까요? https://www.w3schools.com/tags/ev_onsubmit.asp

1005hoon commented on signin/index1.js:
```diff
@@ -75,26 +71,26 @@ function validateUserLoginInfo(e) {
-      const pwRequest = document.querySelector(".pw-request");
-      removeError(pwRequest);
+      const pwErrorMessage = document.querySelector(".pw-error-message");
+      removeError(pwErrorMessage);
```
여전히 `validateUserLoginInfo` 함수가 너무 많은 녀석들을 처리하는것 같아요! 
그리구 지금 e.target.id를 기준으로 실행될 로직을 구분하는건 조금 위험한 로직으로 보입니다. 
각 element에서 validate이 되어야 하는 타이밍에 맞춰 될 수 있도록 이벤트 리스너를 활용해봄은 어떨까요?

```
handleLogin 함수 호출 
- 이메일 값 체크 (이메일 유형인지, 비어있지는 않은지 등)
- 비밀번호 값 체크 (비어있지는 않는지 등)
- 저장된 유저와 같은지 체크 
```
순서로 진행되도록 좀 더 모둘화를 해서 활용할 수 있을것 같습니다.

1005hoon reviewed (changes requested):
코드단위에서 리뷰 남겨두었습니다! 

전반적으로 많이 좋아졌는데요, 조금 더 개선시켜보자면 함수 하나가 해야할 일이 조금 더 명확해지면 좋겠어요. 아직은 조금 로직들이 산재되어있습니다. 이 부분 어려우시다면 디스코드에 남겨주세요! 

그리구 파일의 경우, 같은 파일에 숫자 1을 붙여주신것들이 있는데, 어떤 목적에서 그러신건가요? 
만약 이전 파일들에 대한 기록을 남겨두고 싶으신 거라면 다른 방식을 활용할 수 있을것 같아요. 그게 아니라면 제거해주시는게 좋을것 같습니다.

q45402sk committed:
refactor: 5주차 코드리뷰 반영

- singin/index.js,signup/index.js: 에러메세지 공통 UI 스타일 분리
- singin/index.js,signup/index.js: 함수 분리하기
- singin/index.js,signup/index.js: e.target.id 없애기

1005hoon commented on signin/common.js:
```diff
@@ -26,29 +26,42 @@ function toggleError(messageType, text) {
 
 //눈모양 아이콘
-function pwMasking(input, event, i) {
+function showPassword(input, event, i) {
```
event를 인자로 전달한 목적은 무엇인가요? 함수 내에서 활용되지 않는것 같습니다. 
그리구 showPassword가 해야할 작업은 다음 두가지로 판단되는데요 
1. input 요소의 type을 text로 변환한다
2. input 요소에 속한 eye button의 이미지를 바꾼다

이 때, input 요소와 마찬가지로, 해당 Input 요소에 속한 eye button 요소도 함수의 인자로 전달받아서 index 값인 i를 활용하지 않아도 되도록 하는게 조금 더 좋아보여요

1005hoon reviewed (changes requested):
week5 고생하셨습니다! 뭔가 전반적으로 감을 잡아가는 모습이 보이는데요 ㅎㅎ 

그럼에도 여전히 로직들이 분산되다보니 조금 정신없는 코드 구현이 되어가고 있습니다. 한번 시간을 들여서 코드 전반적인 구조를 정리해보고, 리팩토링을 할 시간이 온것 같아요. 

멘토링 세션도 이제 다음주로 끝나게 될텐데요, 그 전에 시간 가능하시다면 1:1으로 한번 다시 설계부터 구현을 위한 틀을 한번 잡아보는건 어떠실까요? 
좀 부담스러우시다면 여지껏 우리 이약 ㅣ한 내용들, 각 주차별 코드리뷰와 디스코드 스레드에 링크를 다 걸어놓았으니 참고해서 개편해봐도 좋을것 같아요.
